src/main/java/rx/internal/operators/OperatorAll.java

```diff
@@ -61,7 +61,10 @@ public void onNext(T t) {
 
             @Override
             public void onError(Throwable e) {
-                child.onError(e);
+                if (!done) {
+                    done = true;
+                    child.onError(e);
+                }
```

Comment:
Yes, anyplace that would drop the exception due to state/protocol considerations should instead call `RxJavaHooks.onError`.